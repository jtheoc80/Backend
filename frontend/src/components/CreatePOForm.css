.create-po-form {
  max-width: 800px;
  margin: 0 auto;
  padding: 20px;
  background: #fff;
  border-radius: 8px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
}

.form-header {
  margin-bottom: 30px;
  padding-bottom: 15px;
  border-bottom: 2px solid #e9ecef;
}

.form-header h2 {
  margin: 0;
  color: #333;
  font-size: 24px;
  font-weight: 600;
}

.po-form {
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.form-section {
  background: #f8f9fa;
  padding: 20px;
  border-radius: 6px;
  border: 1px solid #e9ecef;
}

.form-section h3 {
  margin: 0 0 20px 0;
  color: #495057;
  font-size: 18px;
  font-weight: 500;
}

.section-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 20px;
}

.section-header h3 {
  margin: 0;
}

.form-row {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 20px;
  margin-bottom: 20px;
}

.form-row:last-child {
  margin-bottom: 0;
}

.form-group {
  display: flex;
  flex-direction: column;
}

.form-group label {
  margin-bottom: 5px;
  color: #495057;
  font-weight: 500;
  font-size: 14px;
}

.form-group input,
.form-group select,
.form-group textarea {
  padding: 10px 12px;
  border: 1px solid #ced4da;
  border-radius: 4px;
  font-size: 14px;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.form-group input:focus,
.form-group select:focus,
.form-group textarea:focus {
  outline: none;
  border-color: #007bff;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
}

.form-group input.error,
.form-group select.error,
.form-group textarea.error {
  border-color: #dc3545;
}

.form-group input:disabled,
.form-group select:disabled,
.form-group textarea:disabled {
  background-color: #e9ecef;
  opacity: 1;
}

.error-message {
  color: #dc3545;
  font-size: 12px;
  margin-top: 4px;
}

.form-error {
  background-color: #f8d7da;
  border: 1px solid #f5c6cb;
  color: #721c24;
  padding: 12px;
  border-radius: 4px;
  margin-bottom: 20px;
}

.item-row {
  background: #fff;
  padding: 15px;
  border-radius: 4px;
  border: 1px solid #dee2e6;
  margin-bottom: 15px;
}

.item-row:last-child {
  margin-bottom: 0;
}

.item-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 15px;
}

.item-header h4 {
  margin: 0;
  color: #495057;
  font-size: 16px;
  font-weight: 500;
}

.total-amount-display {
  display: flex;
  align-items: center;
  background: #e9ecef;
  padding: 10px 12px;
  border-radius: 4px;
  font-size: 16px;
  font-weight: 600;
}

.total-amount-display .currency {
  margin-right: 8px;
  color: #6c757d;
}

.total-amount-display .amount {
  color: #28a745;
}

.form-actions {
  display: flex;
  justify-content: flex-end;
  gap: 10px;
  padding-top: 20px;
  border-top: 1px solid #e9ecef;
}

.btn {
  padding: 10px 20px;
  border: none;
  border-radius: 4px;
  font-size: 14px;
  font-weight: 500;
  cursor: pointer;
  transition: background-color 0.15s ease-in-out;
  text-decoration: none;
  display: inline-block;
  text-align: center;
}

.btn:disabled {
  opacity: 0.65;
  cursor: not-allowed;
}

.btn-primary {
  background-color: #007bff;
  color: white;
}

.btn-primary:hover:not(:disabled) {
  background-color: #0056b3;
}

.btn-secondary {
  background-color: #6c757d;
  color: white;
}

.btn-secondary:hover:not(:disabled) {
  background-color: #545b62;
}

.btn-danger {
  background-color: #dc3545;
  color: white;
}

.btn-danger:hover:not(:disabled) {
  background-color: #c82333;
}

.btn-sm {
  padding: 5px 10px;
  font-size: 12px;
}

/* Responsive Design */
@media (max-width: 768px) {
  .create-po-form {
    padding: 15px;
    margin: 10px;
  }
  
  .form-row {
    grid-template-columns: 1fr;
  }
  
  .section-header {
    flex-direction: column;
    align-items: flex-start;
    gap: 10px;
  }
  
  .form-actions {
    flex-direction: column;
  }
  
  .btn {
    width: 100%;
  }
}