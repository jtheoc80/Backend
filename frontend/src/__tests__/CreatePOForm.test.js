import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import CreatePOForm from '../components/CreatePOForm';

// Mock manufacturers and distributors data
const mockManufacturers = [
  { id: 'mfg001', name: 'Emerson Process Management' },
  { id: 'mfg002', name: 'Kitz Corporation' }
];

const mockDistributors = [
  { id: 'dist001', name: 'Industrial Valve Solutions Inc' },
  { id: 'dist002', name: 'Global Valve Distribution' }
];

describe('CreatePOForm', () => {
  const mockOnSubmit = jest.fn();
  const mockOnCancel = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  const renderComponent = (props = {}) => {
    return render(
      <CreatePOForm
        onSubmit={mockOnSubmit}
        onCancel={mockOnCancel}
        manufacturers={mockManufacturers}
        distributors={mockDistributors}
        {...props}
      />
    );
  };

  describe('Component Rendering', () => {
    it('should render all form sections and fields', () => {
      renderComponent();

      expect(screen.getByRole('heading', { name: 'Create Purchase Order' })).toBeInTheDocument();
      expect(screen.getByLabelText(/PO Number/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Currency/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Manufacturer/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Distributor/)).toBeInTheDocument();
      expect(screen.getByText('Items')).toBeInTheDocument();
      expect(screen.getByText('Total Amount *')).toBeInTheDocument();
      expect(screen.getByLabelText(/Notes/)).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /Create Purchase Order/ })).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /Cancel/ })).toBeInTheDocument();
    });

    it('should render manufacturers and distributors in dropdowns', () => {
      renderComponent();

      const manufacturerSelect = screen.getByLabelText(/Manufacturer/);
      const distributorSelect = screen.getByLabelText(/Distributor/);

      expect(manufacturerSelect).toHaveTextContent('Emerson Process Management');
      expect(manufacturerSelect).toHaveTextContent('Kitz Corporation');
      expect(distributorSelect).toHaveTextContent('Industrial Valve Solutions Inc');
      expect(distributorSelect).toHaveTextContent('Global Valve Distribution');
    });

    it('should render initial item row with all fields', () => {
      renderComponent();

      expect(screen.getByText('Item 1')).toBeInTheDocument();
      expect(screen.getByLabelText(/Valve ID/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Quantity/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Unit Price/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Description/)).toBeInTheDocument();
    });

    it('should display total amount as 0.00 initially', () => {
      renderComponent();

      expect(screen.getByText('0.00')).toBeInTheDocument();
    });
  });

  describe('Form Interactions', () => {
    it('should update input values when user types', async () => {
      const user = userEvent.setup();
      renderComponent();

      const poNumberInput = screen.getByLabelText(/PO Number/);
      await user.type(poNumberInput, 'PO-2024-001');

      expect(poNumberInput).toHaveValue('PO-2024-001');
    });

    it('should update dropdown selections', async () => {
      const user = userEvent.setup();
      renderComponent();

      const manufacturerSelect = screen.getByLabelText(/Manufacturer/);
      await user.selectOptions(manufacturerSelect, 'mfg001');

      expect(manufacturerSelect).toHaveValue('mfg001');
    });

    it('should calculate total amount when item details change', async () => {
      const user = userEvent.setup();
      renderComponent();

      const quantityInput = screen.getByLabelText(/Quantity/);
      const unitPriceInput = screen.getByLabelText(/Unit Price/);

      await user.clear(quantityInput);
      await user.type(quantityInput, '2');
      await user.clear(unitPriceInput);
      await user.type(unitPriceInput, '100.50');

      await waitFor(() => {
        expect(screen.getByText('201.00')).toBeInTheDocument();
      });
    });

    it('should add new item when Add Item button is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      const addItemButton = screen.getByRole('button', { name: /Add Item/ });
      await user.click(addItemButton);

      expect(screen.getByText('Item 2')).toBeInTheDocument();
      expect(screen.getAllByLabelText(/Valve ID/)).toHaveLength(2);
    });

    it('should remove item when Remove button is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      // Add a second item first
      await user.click(screen.getByRole('button', { name: /Add Item/ }));
      expect(screen.getByText('Item 2')).toBeInTheDocument();

      // Remove the second item
      const removeButtons = screen.getAllByRole('button', { name: /Remove/ });
      await user.click(removeButtons[0]);

      expect(screen.queryByText('Item 2')).not.toBeInTheDocument();
    });

    it('should not show remove button when only one item exists', () => {
      renderComponent();

      expect(screen.queryByRole('button', { name: /Remove/ })).not.toBeInTheDocument();
    });
  });

  describe('Form Validation', () => {
    it('should show validation errors for required fields', async () => {
      const user = userEvent.setup();
      renderComponent();

      const submitButton = screen.getByRole('button', { name: /Create Purchase Order/ });
      await user.click(submitButton);

      await waitFor(() => {
        expect(screen.getByText('PO Number is required')).toBeInTheDocument();
        expect(screen.getByText('Manufacturer is required')).toBeInTheDocument();
        expect(screen.getByText('Distributor is required')).toBeInTheDocument();
      });

      expect(mockOnSubmit).not.toHaveBeenCalled();
    });

    it('should validate item fields', async () => {
      const user = userEvent.setup();
      renderComponent();

      // Fill required basic fields
      await user.type(screen.getByLabelText(/PO Number/), 'PO-2024-001');
      await user.selectOptions(screen.getByLabelText(/Manufacturer/), 'mfg001');
      await user.selectOptions(screen.getByLabelText(/Distributor/), 'dist001');

      const submitButton = screen.getByRole('button', { name: /Create Purchase Order/ });
      await user.click(submitButton);

      await waitFor(() => {
        expect(screen.getByText('Valve ID is required')).toBeInTheDocument();
        expect(screen.getByText('Description is required')).toBeInTheDocument();
        expect(screen.getByText('Unit price must be greater than 0')).toBeInTheDocument();
      });
    });

    it('should validate total amount is greater than 0', async () => {
      const user = userEvent.setup();
      renderComponent();

      // Fill all required fields but leave amounts as 0
      await user.type(screen.getByLabelText(/PO Number/), 'PO-2024-001');
      await user.selectOptions(screen.getByLabelText(/Manufacturer/), 'mfg001');
      await user.selectOptions(screen.getByLabelText(/Distributor/), 'dist001');
      await user.type(screen.getByLabelText(/Valve ID/), 'valve-001');
      await user.type(screen.getByLabelText(/Description/), 'Test valve');

      const submitButton = screen.getByRole('button', { name: /Create Purchase Order/ });
      await user.click(submitButton);

      await waitFor(() => {
        expect(screen.getByText('Total amount must be greater than 0')).toBeInTheDocument();
      });
    });

    it('should clear validation errors when user corrects input', async () => {
      const user = userEvent.setup();
      renderComponent();

      // Trigger validation error
      const submitButton = screen.getByRole('button', { name: /Create Purchase Order/ });
      await user.click(submitButton);

      await waitFor(() => {
        expect(screen.getByText('PO Number is required')).toBeInTheDocument();
      });

      // Fix the error
      const poNumberInput = screen.getByLabelText(/PO Number/);
      await user.type(poNumberInput, 'PO-2024-001');

      await waitFor(() => {
        expect(screen.queryByText('PO Number is required')).not.toBeInTheDocument();
      });
    });
  });

  describe('Form Submission', () => {
    const fillValidForm = async (user) => {
      await user.type(screen.getByLabelText(/PO Number/), 'PO-2024-001');
      await user.selectOptions(screen.getByLabelText(/Manufacturer/), 'mfg001');
      await user.selectOptions(screen.getByLabelText(/Distributor/), 'dist001');
      await user.type(screen.getByLabelText(/Valve ID/), 'valve-001');
      await user.clear(screen.getByLabelText(/Quantity/));
      await user.type(screen.getByLabelText(/Quantity/), '2');
      await user.type(screen.getByLabelText(/Unit Price/), '100.50');
      await user.type(screen.getByLabelText(/Description/), 'Ball Valve 1/2 inch');
      await user.type(screen.getByLabelText(/Notes/), 'Urgent order');
    };

    it('should submit form with correct data when all fields are valid', async () => {
      const user = userEvent.setup();
      mockOnSubmit.mockResolvedValue({});
      renderComponent();

      await fillValidForm(user);

      const submitButton = screen.getByRole('button', { name: /Create Purchase Order/ });
      await user.click(submitButton);

      await waitFor(() => {
        expect(mockOnSubmit).toHaveBeenCalledWith({
          po_number: 'PO-2024-001',
          manufacturer_id: 'mfg001',
          distributor_id: 'dist001',
          total_amount: 201,
          currency: 'USD',
          items: [{
            valve_id: 'valve-001',
            quantity: 2,
            unit_price: 100.50,
            description: 'Ball Valve 1/2 inch'
          }],
          notes: 'Urgent order'
        });
      });
    });

    it('should show loading state during submission', async () => {
      const user = userEvent.setup();
      let resolveSubmit;
      mockOnSubmit.mockImplementation(() => new Promise(resolve => {
        resolveSubmit = resolve;
      }));
      renderComponent();

      await fillValidForm(user);

      const submitButton = screen.getByRole('button', { name: /Create Purchase Order/ });
      await user.click(submitButton);

      await waitFor(() => {
        expect(screen.getByText('Creating...')).toBeInTheDocument();
        expect(submitButton).toBeDisabled();
      });

      resolveSubmit({});
    });

    it('should handle submission errors', async () => {
      const user = userEvent.setup();
      mockOnSubmit.mockRejectedValue(new Error('Submission failed'));
      renderComponent();

      await fillValidForm(user);

      const submitButton = screen.getByRole('button', { name: /Create Purchase Order/ });
      await user.click(submitButton);

      await waitFor(() => {
        expect(screen.getByText('Failed to create purchase order. Please try again.')).toBeInTheDocument();
      });
    });
  });

  describe('Component Callbacks', () => {
    it('should call onCancel when Cancel button is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      const cancelButton = screen.getByRole('button', { name: /Cancel/ });
      await user.click(cancelButton);

      expect(mockOnCancel).toHaveBeenCalledTimes(1);
    });
  });

  describe('Props Handling', () => {
    it('should handle empty manufacturers and distributors arrays', () => {
      renderComponent({ manufacturers: [], distributors: [] });

      const manufacturerSelect = screen.getByLabelText(/Manufacturer/);
      const distributorSelect = screen.getByLabelText(/Distributor/);

      expect(manufacturerSelect).toHaveTextContent('Select a manufacturer');
      expect(distributorSelect).toHaveTextContent('Select a distributor');
    });

    it('should handle missing props gracefully', () => {
      render(
        <CreatePOForm
          onSubmit={mockOnSubmit}
          onCancel={mockOnCancel}
        />
      );

      expect(screen.getByRole('heading', { name: 'Create Purchase Order' })).toBeInTheDocument();
    });
  });

  describe('Accessibility', () => {
    it('should have proper form labels and structure', () => {
      renderComponent();

      expect(screen.getByLabelText(/PO Number/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Manufacturer/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Distributor/)).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /Create Purchase Order/ })).toBeInTheDocument();
    });

    it('should support keyboard navigation', () => {
      renderComponent();

      const firstInput = screen.getByLabelText(/PO Number/);
      firstInput.focus();
      expect(firstInput).toHaveFocus();
    });
  });

  describe('Snapshot Testing', () => {
    it('should match snapshot with default props', () => {
      const { container } = renderComponent();
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with multiple items', async () => {
      const user = userEvent.setup();
      const { container } = renderComponent();

      await user.click(screen.getByRole('button', { name: /Add Item/ }));
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with validation errors', async () => {
      const user = userEvent.setup();
      const { container } = renderComponent();

      const submitButton = screen.getByRole('button', { name: /Create Purchase Order/ });
      await user.click(submitButton);

      await waitFor(() => {
        expect(screen.getByText('PO Number is required')).toBeInTheDocument();
      });

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});