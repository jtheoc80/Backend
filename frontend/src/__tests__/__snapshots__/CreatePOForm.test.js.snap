// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePOForm Snapshot Testing should match snapshot with default props 1`] = `
<div
  class="create-po-form"
>
  <div
    class="form-header"
  >
    <h2>
      Create Purchase Order
    </h2>
  </div>
  <form
    class="po-form"
  >
    <div
      class="form-section"
    >
      <h3>
        Basic Information
      </h3>
      <div
        class="form-row"
      >
        <div
          class="form-group"
        >
          <label
            for="po_number"
          >
            PO Number *
          </label>
          <input
            class=""
            id="po_number"
            name="po_number"
            placeholder="e.g., PO-2024-001"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="currency"
          >
            Currency
          </label>
          <select
            id="currency"
            name="currency"
          >
            <option
              value="USD"
            >
              USD
            </option>
            <option
              value="EUR"
            >
              EUR
            </option>
            <option
              value="GBP"
            >
              GBP
            </option>
          </select>
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="form-group"
        >
          <label
            for="manufacturer_id"
          >
            Manufacturer *
          </label>
          <select
            class=""
            id="manufacturer_id"
            name="manufacturer_id"
          >
            <option
              value=""
            >
              Select a manufacturer
            </option>
            <option
              value="mfg001"
            >
              Emerson Process Management
            </option>
            <option
              value="mfg002"
            >
              Kitz Corporation
            </option>
          </select>
        </div>
        <div
          class="form-group"
        >
          <label
            for="distributor_id"
          >
            Distributor *
          </label>
          <select
            class=""
            id="distributor_id"
            name="distributor_id"
          >
            <option
              value=""
            >
              Select a distributor
            </option>
            <option
              value="dist001"
            >
              Industrial Valve Solutions Inc
            </option>
            <option
              value="dist002"
            >
              Global Valve Distribution
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="form-section"
    >
      <div
        class="section-header"
      >
        <h3>
          Items
        </h3>
        <button
          class="btn btn-secondary"
          type="button"
        >
          Add Item
        </button>
      </div>
      <div
        class="item-row"
      >
        <div
          class="item-header"
        >
          <h4>
            Item 
            1
          </h4>
        </div>
        <div
          class="form-row"
        >
          <div
            class="form-group"
          >
            <label
              for="valve_id_0"
            >
              Valve ID *
            </label>
            <input
              class=""
              id="valve_id_0"
              placeholder="e.g., valve-001"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="quantity_0"
            >
              Quantity *
            </label>
            <input
              class=""
              id="quantity_0"
              min="1"
              type="number"
              value="1"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="unit_price_0"
            >
              Unit Price *
            </label>
            <input
              class=""
              id="unit_price_0"
              min="0"
              step="0.01"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            for="description_0"
          >
            Description *
          </label>
          <input
            class=""
            id="description_0"
            placeholder="e.g., Ball Valve 1/2 inch"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="form-section"
    >
      <div
        class="form-group"
      >
        <label
          for="total_amount"
        >
          Total Amount *
        </label>
        <div
          class="total-amount-display"
        >
          <span
            class="currency"
          >
            USD
          </span>
          <span
            class="amount"
          >
            0.00
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          for="notes"
        >
          Notes
        </label>
        <textarea
          id="notes"
          name="notes"
          placeholder="Additional notes or special instructions..."
          rows="3"
        />
      </div>
    </div>
    <div
      class="form-actions"
    >
      <button
        class="btn btn-secondary"
        type="button"
      >
        Cancel
      </button>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Create Purchase Order
      </button>
    </div>
  </form>
</div>
`;

exports[`CreatePOForm Snapshot Testing should match snapshot with multiple items 1`] = `
<div
  class="create-po-form"
>
  <div
    class="form-header"
  >
    <h2>
      Create Purchase Order
    </h2>
  </div>
  <form
    class="po-form"
  >
    <div
      class="form-section"
    >
      <h3>
        Basic Information
      </h3>
      <div
        class="form-row"
      >
        <div
          class="form-group"
        >
          <label
            for="po_number"
          >
            PO Number *
          </label>
          <input
            class=""
            id="po_number"
            name="po_number"
            placeholder="e.g., PO-2024-001"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="currency"
          >
            Currency
          </label>
          <select
            id="currency"
            name="currency"
          >
            <option
              value="USD"
            >
              USD
            </option>
            <option
              value="EUR"
            >
              EUR
            </option>
            <option
              value="GBP"
            >
              GBP
            </option>
          </select>
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="form-group"
        >
          <label
            for="manufacturer_id"
          >
            Manufacturer *
          </label>
          <select
            class=""
            id="manufacturer_id"
            name="manufacturer_id"
          >
            <option
              value=""
            >
              Select a manufacturer
            </option>
            <option
              value="mfg001"
            >
              Emerson Process Management
            </option>
            <option
              value="mfg002"
            >
              Kitz Corporation
            </option>
          </select>
        </div>
        <div
          class="form-group"
        >
          <label
            for="distributor_id"
          >
            Distributor *
          </label>
          <select
            class=""
            id="distributor_id"
            name="distributor_id"
          >
            <option
              value=""
            >
              Select a distributor
            </option>
            <option
              value="dist001"
            >
              Industrial Valve Solutions Inc
            </option>
            <option
              value="dist002"
            >
              Global Valve Distribution
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="form-section"
    >
      <div
        class="section-header"
      >
        <h3>
          Items
        </h3>
        <button
          class="btn btn-secondary"
          type="button"
        >
          Add Item
        </button>
      </div>
      <div
        class="item-row"
      >
        <div
          class="item-header"
        >
          <h4>
            Item 
            1
          </h4>
          <button
            class="btn btn-danger btn-sm"
            type="button"
          >
            Remove
          </button>
        </div>
        <div
          class="form-row"
        >
          <div
            class="form-group"
          >
            <label
              for="valve_id_0"
            >
              Valve ID *
            </label>
            <input
              class=""
              id="valve_id_0"
              placeholder="e.g., valve-001"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="quantity_0"
            >
              Quantity *
            </label>
            <input
              class=""
              id="quantity_0"
              min="1"
              type="number"
              value="1"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="unit_price_0"
            >
              Unit Price *
            </label>
            <input
              class=""
              id="unit_price_0"
              min="0"
              step="0.01"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            for="description_0"
          >
            Description *
          </label>
          <input
            class=""
            id="description_0"
            placeholder="e.g., Ball Valve 1/2 inch"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="item-row"
      >
        <div
          class="item-header"
        >
          <h4>
            Item 
            2
          </h4>
          <button
            class="btn btn-danger btn-sm"
            type="button"
          >
            Remove
          </button>
        </div>
        <div
          class="form-row"
        >
          <div
            class="form-group"
          >
            <label
              for="valve_id_1"
            >
              Valve ID *
            </label>
            <input
              class=""
              id="valve_id_1"
              placeholder="e.g., valve-001"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="quantity_1"
            >
              Quantity *
            </label>
            <input
              class=""
              id="quantity_1"
              min="1"
              type="number"
              value="1"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="unit_price_1"
            >
              Unit Price *
            </label>
            <input
              class=""
              id="unit_price_1"
              min="0"
              step="0.01"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            for="description_1"
          >
            Description *
          </label>
          <input
            class=""
            id="description_1"
            placeholder="e.g., Ball Valve 1/2 inch"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="form-section"
    >
      <div
        class="form-group"
      >
        <label
          for="total_amount"
        >
          Total Amount *
        </label>
        <div
          class="total-amount-display"
        >
          <span
            class="currency"
          >
            USD
          </span>
          <span
            class="amount"
          >
            0.00
          </span>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          for="notes"
        >
          Notes
        </label>
        <textarea
          id="notes"
          name="notes"
          placeholder="Additional notes or special instructions..."
          rows="3"
        />
      </div>
    </div>
    <div
      class="form-actions"
    >
      <button
        class="btn btn-secondary"
        type="button"
      >
        Cancel
      </button>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Create Purchase Order
      </button>
    </div>
  </form>
</div>
`;

exports[`CreatePOForm Snapshot Testing should match snapshot with validation errors 1`] = `
<div
  class="create-po-form"
>
  <div
    class="form-header"
  >
    <h2>
      Create Purchase Order
    </h2>
  </div>
  <form
    class="po-form"
  >
    <div
      class="form-section"
    >
      <h3>
        Basic Information
      </h3>
      <div
        class="form-row"
      >
        <div
          class="form-group"
        >
          <label
            for="po_number"
          >
            PO Number *
          </label>
          <input
            class="error"
            id="po_number"
            name="po_number"
            placeholder="e.g., PO-2024-001"
            type="text"
            value=""
          />
          <span
            class="error-message"
          >
            PO Number is required
          </span>
        </div>
        <div
          class="form-group"
        >
          <label
            for="currency"
          >
            Currency
          </label>
          <select
            id="currency"
            name="currency"
          >
            <option
              value="USD"
            >
              USD
            </option>
            <option
              value="EUR"
            >
              EUR
            </option>
            <option
              value="GBP"
            >
              GBP
            </option>
          </select>
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="form-group"
        >
          <label
            for="manufacturer_id"
          >
            Manufacturer *
          </label>
          <select
            class="error"
            id="manufacturer_id"
            name="manufacturer_id"
          >
            <option
              value=""
            >
              Select a manufacturer
            </option>
            <option
              value="mfg001"
            >
              Emerson Process Management
            </option>
            <option
              value="mfg002"
            >
              Kitz Corporation
            </option>
          </select>
          <span
            class="error-message"
          >
            Manufacturer is required
          </span>
        </div>
        <div
          class="form-group"
        >
          <label
            for="distributor_id"
          >
            Distributor *
          </label>
          <select
            class="error"
            id="distributor_id"
            name="distributor_id"
          >
            <option
              value=""
            >
              Select a distributor
            </option>
            <option
              value="dist001"
            >
              Industrial Valve Solutions Inc
            </option>
            <option
              value="dist002"
            >
              Global Valve Distribution
            </option>
          </select>
          <span
            class="error-message"
          >
            Distributor is required
          </span>
        </div>
      </div>
    </div>
    <div
      class="form-section"
    >
      <div
        class="section-header"
      >
        <h3>
          Items
        </h3>
        <button
          class="btn btn-secondary"
          type="button"
        >
          Add Item
        </button>
      </div>
      <div
        class="item-row"
      >
        <div
          class="item-header"
        >
          <h4>
            Item 
            1
          </h4>
        </div>
        <div
          class="form-row"
        >
          <div
            class="form-group"
          >
            <label
              for="valve_id_0"
            >
              Valve ID *
            </label>
            <input
              class="error"
              id="valve_id_0"
              placeholder="e.g., valve-001"
              type="text"
              value=""
            />
            <span
              class="error-message"
            >
              Valve ID is required
            </span>
          </div>
          <div
            class="form-group"
          >
            <label
              for="quantity_0"
            >
              Quantity *
            </label>
            <input
              class=""
              id="quantity_0"
              min="1"
              type="number"
              value="1"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="unit_price_0"
            >
              Unit Price *
            </label>
            <input
              class="error"
              id="unit_price_0"
              min="0"
              step="0.01"
              type="number"
              value=""
            />
            <span
              class="error-message"
            >
              Unit price must be greater than 0
            </span>
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            for="description_0"
          >
            Description *
          </label>
          <input
            class="error"
            id="description_0"
            placeholder="e.g., Ball Valve 1/2 inch"
            type="text"
            value=""
          />
          <span
            class="error-message"
          >
            Description is required
          </span>
        </div>
      </div>
    </div>
    <div
      class="form-section"
    >
      <div
        class="form-group"
      >
        <label
          for="total_amount"
        >
          Total Amount *
        </label>
        <div
          class="total-amount-display"
        >
          <span
            class="currency"
          >
            USD
          </span>
          <span
            class="amount"
          >
            0.00
          </span>
        </div>
        <span
          class="error-message"
        >
          Total amount must be greater than 0
        </span>
      </div>
      <div
        class="form-group"
      >
        <label
          for="notes"
        >
          Notes
        </label>
        <textarea
          id="notes"
          name="notes"
          placeholder="Additional notes or special instructions..."
          rows="3"
        />
      </div>
    </div>
    <div
      class="form-actions"
    >
      <button
        class="btn btn-secondary"
        type="button"
      >
        Cancel
      </button>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Create Purchase Order
      </button>
    </div>
  </form>
</div>
`;
